packages/disploy/src/structs/Modal.ts

```diff
@@ -0,0 +1,33 @@
+import type { ActionRow } from "./ActionRow";
+
+export class Modal {
+    public customId?: string;
+    public title?: string;
+    public components: ActionRow[] = [];
+
+    setCustomId(id: string){
+        this.customId = id;
+        return this;
+    }
+
+    setTitle(title: string){
+        this.title = title;
+        return this;
+    }
+
+    setComponents(components: ActionRow[]){
+        this.components = components;
+        return this;
+    }
+
+    addComponents(components: ActionRow[]){
+        this.components.push(...components);
+        return this;
+    }
+
+    toJSON(){
+        return JSON.stringify(this);
+    }
+
+    constructor() {}
```

Comment:
Don't force users to use the builder pattern, let them pass an object to fill the class.